reactivesocket-client/src/main/java/io/reactivesocket/client/LoadBalancer.java

```diff
@@ -121,18 +174,43 @@ public LoadBalancer(Publisher<List<ReactiveSocketFactory<SocketAddress>>> factor
     }
 
     private synchronized void addSockets(int numberOfNewSocket) {
-        activeFactories.entrySet()
-            .stream()
-            // available factories that don't map to an already established socket
-            .filter(e -> !activeSockets.containsKey(e.getKey()))
-            .map(e -> e.getValue())
-            .filter(factory -> factory.availability() > 0.0)
-            .sorted((a, b) -> -Double.compare(a.availability(), b.availability()))
-            .limit(numberOfNewSocket)
-            .forEach(factory -> {
-                pendingSockets += 1;
-                factory.apply().subscribe(new SocketAdder(factory.remote()));
-            });
+        int n = Math.min(activeFactories.size(), numberOfNewSocket);
+        int maxEffort = 1 + activeFactories.size()/2;
+        int effort = 0;
+        while (n > 0 && effort < maxEffort) {
```

Comment:
(I remember we discussing it offline, so if this is in your TODOs, then let me know)

If the number of hosts < aperture size, should we favor creating more connections on the same host?